### PR TITLE
Wasm

### LINKED ISSUE #5: High-level API: LrsExt
- [x] from LrmMeasure to coordinate
- [x] from coordinate and lrm id to LrmMeasure
- [ ] from coordinate to list of (lrm id, LrmMeasure)

### LINKED ISSUE #7: Web app demo
- Visualize the data
- Transform coordinate to LrmMeasures
- Transform LrmMeasures to coordinates

### PR DESCRIPTION
**Pull request by Tristramg**
Closes #7 
Closes #5

### CODE REVIEW

File: html_demo/index.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+    map = new maplibregl.Map({
+        container: 'map', // container id
+        style: 'https://api.maptiler.com/maps/basic-v2/style.json?key=dvUgavMVX1luWp9QqjS0',
+        center: [2.3469, 46.8589], // starting position [lng, lat]
+        zoom: 5, // starting zoom,
+    });
+}
+
+init();
```
Comment (Khoyo) [inline html_demo/index.js:72]: Missing newline at EOF ;)

File: html_demo/index.html [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+            <h1>Démo LRS</h1>
+            <input type="file" />
+            <div id="output"></div>
+        </div>
+        <div id="map">
+        </div>
+    </main>
+</body>
+
+</html>
```
Comment (Khoyo) [inline html_demo/index.html:42]: Idem

File: README.md [hunk 3]
```diff
@@ -1,6 +1,52 @@
...
 
-TODO: explain what this library is for
+[Linear Reference](https://en.wikipedia.org/wiki/Linear_referencing) allows to identify a position along a curve (a road, a canal, a railroad…) relative to fixed reference points. 
+
+We call:
+- **anchor** a reference point: it can be milestones, a landmark, an intersection…
+- **scale** the list of _anchors_ and the theoretical distance between them,
+- **curve** the physical description of the object to reference,
+- **linear referencing method** (LRM) the combination of a _scale_ and a _curve_,
+- **linear referencing method** (LRS) the complete set of data.
```
Comment (JulesPeignier) [inline README.md:10]: **linear referencing system** (LRS) the complete set of data.

**Review by JulesPeignier — APPROVED** [review]

File: README.md (hunk 3 above)
Comment (Tristramg) [reply README.md]: good catch! thank you

File: README.md [hunk 4]
```diff
@@ -1,6 +1,52 @@
...
+
+We call:
+- **anchor** a reference point: it can be milestones, a landmark, an intersection…
+- **scale** the list of _anchors_ and the theoretical distance between them,
+- **curve** the physical description of the object to reference,
+- **linear referencing method** (LRM) the combination of a _scale_ and a _curve_,
+- **linear referencing system** (LRS) the complete set of data.
+
+
+While the logic is quite simple, multiple small subtilites make the LRS difficulte to use.
```
Comment (louisgreiner) [inline README.md:13]: typos:
- subtilites -> subtleties
- difficulte -> difficult

File: README.md [hunk 5]
```diff
@@ -1,6 +1,52 @@
...
+Distances between milestone change (construction of a bypass around a town), the origin of curve is displaced (the railwail station moved), there is only one scale for a river and it’s parallel running canal…
```
Comment (louisgreiner) [inline README.md:14]: typo : it's -> its

File: README.md [hunk 6]
```diff
@@ -1,6 +1,52 @@
...
+
+This library aims to handle many edge cases and makes little assumptions about the data:
+- anchors are not always numbers,
+- anchors doesn’t need to be on the curve,
```
Comment (louisgreiner) [inline README.md:18]: typo: doesn't -> don't

File: src/curves.rs [hunk 7]
```diff
@@ -54,6 +55,12 @@ pub trait Curve {
     /// TODO: implement return of all the points of intersection.
     /// When the segment is collinear with the [`Curve`] it is ignored.
     fn intersect_segment(&self, segment: Line) -> Option<Point>;
+
+    /// Get the geometry of the `Curve`
+    fn as_linestring(&self) -> LineString;
+
+    /// Get a range of the `Curvre`
```
Comment (louisgreiner) [inline src/curves.rs:62]: typo: `Curvre` -> `Curve`

File: src/lrs.rs [hunk 8]
```diff
@@ -181,24 +196,30 @@ impl<CurveImpl: Curve> Lrs<CurveImpl> {
             });
         }
 
+        println!("iterations on lrm scales");
```
Comment (louisgreiner) [inline src/lrs.rs:199]: is it intended to keep this print ? if yes, ignore this comment

**Review by louisgreiner — COMMENTED** [review]
Only a few changes on the comments. Thank you for debugging the SphericalLineStringCurve at the same time ;) Well done

File: src/lrs.rs (hunk 8 above)
Comment (Tristramg) [reply src/lrs.rs:199]: Not, not at all, thank you :)

**Comment by Tristramg** [comment]
Thank you for the reviews! :rocket:

**Pull request merged**